src/components/cards/ConstructBuilding.tsx

```diff
@@ -1,123 +0,0 @@
-import React, { Dispatch, SetStateAction } from "react";
-import {
-  Buildings,
-  BuildingType,
-  ResourcePool,
-  Resources,
-  ResourceType,
-} from "../../types/";
-import {
-  cloneBasicObjectWithJSON,
-  updateResources as updateResourcePool,
-} from "../../utils";
-import { resourceChecker } from "../../utils/resourceChecker";
-import { AddRemoveButton } from "../buttons";
-import {
-  CardCostsInfo,
-  CardHeader,
-  CardShowCount,
-  CardSymbol,
-  CardTemplate,
-} from "../cards";
-import CardImageHeaderDescriptionContainer from "./CardImageHeaderDescriptionContainer";
-
-interface ConstructBuildingProps {
-  buildings: Buildings;
-  setBuildings: Dispatch<SetStateAction<Buildings>>;
-  /* setBuildings: (buildings: Buildings) => void; */
-  /* FIXME: Why is this being inferred as a number? */
-  buildingType: BuildingType;
-  resources: Resources;
-  setResources: Dispatch<SetStateAction<Resources>>;
-  resourcePool: ResourcePool;
-  setResourcePool: Dispatch<SetStateAction<ResourcePool>>;
-  /* setResources: (resources: Resources) => void; */
-}
-
-export default function ConstructBuilding({
-  buildings,
-  setBuildings,
-  buildingType,
-  resources,
-  setResources,
-  resourcePool,
-  setResourcePool,
-}: ConstructBuildingProps) {
-  // get the resource costs for the given building type
-  const costsObject = buildings[buildingType].resourceCosts;
-
-  const handleCancelClick = (buildingType: string) => {
-    if (buildings[buildingType].underConstruction === true) {
-      // cancel building construction
-      const clonedBuildingsData = cloneBasicObjectWithJSON(buildings);
-      clonedBuildingsData[buildingType as BuildingType].underConstruction =
-        false;
-      setBuildings(clonedBuildingsData);
-
-      // give back resources
-      const clonedResourcePool = cloneBasicObjectWithJSON(resourcePool);
-      updateResourcePool(-1, costsObject, clonedResourcePool);
-
-      setResourcePool(clonedResourcePool);
-    }
-  };
-
-  const handleBuildClick = (buildingType: string) => {
-    // check that you've collected all required resources
-    const resourceCheck = resourceChecker(1, costsObject, resourcePool);
-
-    if (buildings[buildingType].underConstruction === false && resourceCheck) {
-      // set the building to be constructed
-      const clonedBuildingsData = cloneBasicObjectWithJSON(buildings);
-      clonedBuildingsData[buildingType].underConstruction = true;
-      setBuildings(clonedBuildingsData);
```

Comment:
Ok I think I got a better idea of whats happening here. You want to to set a specific building's `underConstruction` to true and then update that in the `buildings` state.

So what you can do instead of cloning here, is pass the newly updated object back to the parent via a function. The function implementation in the parent will handle creating a new buildings state and only update the one particular building. It may seem like more work to add another function here, but you can be as specific as you want. You will also enjoy types if you set it up correctly. And you wouldnt be passing in `buildings` if you don't need it.

We may have to go over this in a little more detail next session.